kie-wb-smoke-tests/src/test/java/org/kie/smoke/wb/rest/GuvnorRestSmokeIntegrationTest.java

```diff
@@ -251,65 +266,70 @@ public void testManipulatingRepositoriesAndProjects() throws Exception {
+                    202, user, password, 
+                    project, 
+                    CreateProjectRequest.class);
+                
+            // rest/jobs/{jobId} GET
+            waitForJobToComplete(deploymentUrl, createProjectRequest.getJobId(), createProjectRequest.getStatus());
+        }
 
-        // rest/jobs/{jobId} GET
-        waitForJobToComplete(deploymentUrl, createProjectRequest.getJobId(), createProjectRequest.getStatus(), requestHelper);
+        {
+            // rest/repositories/{repoName}/projects POST
+            CompileProjectRequest compileRequest = post(deploymentUrl,
+                    "rest/repositories/" + repoName + "/projects/" + projectName + "/maven/compile", mediaType,
+                    202, user, password, 
+                    CompileProjectRequest.class);
 
-        // TODO implement GET
-        // rest/repositories/{repoName}/projects GET
-        /** get projects, compare/verify that new project is in list **/
+            // rest/jobs/{jobId} GET
+            waitForJobToComplete(deploymentUrl, compileRequest.getJobId(), compileRequest.getStatus());
+        }
 
-        // TODO implement DELETE
-        // rest/repositories/{repoName}/projects DELETE
-        /** delete projects, verify that list of projects is now one less */
     }
 
-    private void waitForJobToComplete(URL deploymentUrl, String jobId, JobStatus jobStatus, RestRequestHelper requestHelper) throws Exception {
-        waitForJobToComplete(deploymentUrl, jobId, jobStatus, requestHelper, JobStatus.SUCCESS);
+    private JobResult waitForJobToComplete(URL deploymentUrl, String jobId, JobStatus jobStatus) throws Exception {
+        return waitForJobToHaveStatus(deploymentUrl, jobId, jobStatus, JobStatus.SUCCESS);
     }
     
-    private void waitForJobToComplete(URL deploymentUrl, String jobId, JobStatus jobStatus, RestRequestHelper requestHelper, JobStatus expectedStatus) throws Exception {
-        assertEquals("Initial status of request", JobStatus.APPROVED, jobStatus);
-        
+    private JobResult waitForJobToHaveStatus(URL deploymentUrl, String jobId, JobStatus jobStatus, JobStatus expectedStatus ) throws Exception {
+        assertTrue( "Initial status of request should be ACCEPTED or APROVED: " + jobStatus, 
+                jobStatus.equals(JobStatus.ACCEPTED) || jobStatus.equals(JobStatus.APPROVED) );
         int wait = 0;
-        do {
-            Thread.sleep(jobCompleteSleepSecs * 1000);
-            ClientRequest restRequest = requestHelper.createRequest("jobs/" + jobId);
-            JobResult jobResult = get(restRequest, mediaType, JobResult.class);
-            logger.debug("]] " + convertObjectToJsonString(jobResult));
-            assertEquals(jobResult.getJobId(), jobId);
+        JobResult jobResult = null;
+        while( ! jobStatus.equals(JobStatus.SUCCESS) && wait < maxTries ) {
```

Comment:
In some cases the expected status could be different than the SUCCESS, for example when checking if the server correctly returns NOT_FOUND for resources that does not exists. 

Could we check here that the status changed from the initial APPROVED or ACCEPTED to some "final" state (e.g. the NOT_FOUND or SUCCESS).